### PR TITLE
🛃 Analyze Entries - Improvements

### PR DESCRIPTION
**Pull request by Gilad-Gur-Andelman**

### CODE REVIEW

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/105aff39-78b8-4a79-840c-a450755a758a/artifacts/0/out/make/Loadmill-3.1.8-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/105aff39-78b8-4a79-840c-a450755a758a/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.8.zip)

**Comment by Gilad-Gur-Andelman** [comment]
Windows Build: 1. [Setup.exe](https://output.circle-artifacts.com/output/job/f8f6080e-a291-47f1-9915-583389f057e3/artifacts/0/out/make/squirrel.windows/x64/RELEASES) 2. [.zip](https://output.circle-artifacts.com/output/job/f8f6080e-a291-47f1-9915-583389f057e3/artifacts/0/out/make/zip/win32/x64/Loadmill-win32-x64-3.1.8.zip)

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/de166d96-8382-4573-b0ec-2131179960fb/artifacts/0/out/make/Loadmill-3.1.8-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/de166d96-8382-4573-b0ec-2131179960fb/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.8.zip)

**Comment by Gilad-Gur-Andelman** [comment]
Windows Build: 1. [Setup.exe](https://output.circle-artifacts.com/output/job/dd722326-4594-4325-a33b-55148d42314d/artifacts/0/out/make/squirrel.windows/x64/RELEASES) 2. [.zip](https://output.circle-artifacts.com/output/job/dd722326-4594-4325-a33b-55148d42314d/artifacts/0/out/make/zip/win32/x64/Loadmill-win32-x64-3.1.8.zip)

File: src/main-process/fetch/index.ts [hunk 1]
```diff
@@ -8,7 +9,7 @@ const httpsAgent = new https.Agent({
 
 const _fetch = async (path: string, reqInit: RequestInit = {}): Promise<Response> => {
   const response = await fetch(path, {
-    agent: httpsAgent,
+    agent: app.isPackaged && httpsAgent,
```
Comment (yigaldviri) [inline src/main-process/fetch/index.ts:12]: can you explain this one?

File: src/main-process/call-loadmill-api.ts [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { fetch, RequestInit, Response } from './fetch';
 
 export const callLoadmillApi = async (path: string, init: RequestInit = {}): Promise<Response> => {
   const cookie = await getCookie();
+  if (!cookie) {
+    throw new Error('No cookie');
```
Comment (yigaldviri) [inline src/main-process/call-loadmill-api.ts:10]: Not sure what the user should do with that message

Comment (yigaldviri) [reply src/main-process/call-loadmill-api.ts:10]: I see you change it later

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/325c612e-44fd-4700-9bb4-63b1643ac072/artifacts/0/out/make/Loadmill-3.1.8-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/325c612e-44fd-4700-9bb4-63b1643ac072/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.8.zip)

**Comment by Gilad-Gur-Andelman** [comment]
Windows Build: 1. [Setup.exe](https://output.circle-artifacts.com/output/job/8e27782a-03c3-4504-a4ad-f42d0b22a344/artifacts/0/out/make/squirrel.windows/x64/RELEASES) 2. [.zip](https://output.circle-artifacts.com/output/job/8e27782a-03c3-4504-a4ad-f42d0b22a344/artifacts/0/out/make/zip/win32/x64/Loadmill-win32-x64-3.1.8.zip)

File: src/main-process/fetch/index.ts (hunk 1 above)
Comment (Gilad-Gur-Andelman) [reply src/main-process/fetch/index.ts:12]: When we are in dev env and pointing at the local server, the `httpsAgent` fails requests to `http`.
So `isPackaged` is true only when not in dev mode.

**Review by onniaa — APPROVED** [review]

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/e62e10c9-18f5-4f1f-b1c3-6cbc62e96b93/artifacts/0/out/make/Loadmill-3.1.8-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/e62e10c9-18f5-4f1f-b1c3-6cbc62e96b93/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.8.zip)

**Pull request merged**